Reorder inheritance of RemoveBadChannelsRecording and the arguments.

maurotoro opened the pull request:
If a recording was passed to the remove_bad_channel preprocessor, the resulting object got the recording and probe properties destroyed. Reordering the inheritance and changing some arguments seems to correct this issue.

samuelgarcia commented:
Thanks. This looks better.
Could we add a very short test for test to be sure ?

maurotoro commented:
I added more assert tests on the original, comparing the number of segments, their lengths and location of the channels that survived. By mistake, I also added some noise to the .gitignore, can I somehow keep those apart only to my fork? Suerte!

samuelgarcia commented:
Perfect.
**.gitignore** is ok, you will be back soon here I hope.
muchísimas gracias.
